core/store/src/flat/metrics.rs

```diff
@@ -50,3 +51,88 @@ impl FlatStorageMetrics {
         self.cached_changes_size.set(cached_changes_size as i64);
     }
 }
+
+/// Metrics for flat storage resharding.
+///
+/// This struct is a collection of metrics to monitor the operation of splitting a shard.
+pub struct FlatStorageReshardingShardSplitMetrics {
+    parent_shard: ShardUId,
+    left_child_shard: ShardUId,
+    right_child_shard: ShardUId,
+    parent_status: IntGauge,
```

Comment:
Is this the standard way for enum like metrics? Would be nicer to have some sort of enum-gauge here if prometheus supports that. Not a biggie, this obviously works.

Reply:
I copied the approach from older metrics.. seems to do the job fine as you said.